merge_checks/runner.py

```diff
@@ -32,8 +32,11 @@ class State(Enum):
 def get_commit_and_status() -> tuple[Commit, Status]:
     commit = Commit(
         repository=os.environ["GITHUB_REPOSITORY"],
-        commit_sha=os.environ["GITHUB_SHA"],
         token=os.environ["GITHUB_TOKEN"],
+        # TODO: Set to os.environ["GITHUB_HEAD_SHA"] without fallback, after merge-checks.yml has been updated
```

Comment:
Oh, I didn't remember anymore that it was a standard environment variable and was **not** passed by the workflow. Yes, in this case, renaming is a good thing. But I would add a comment to the code to justify it, otherwise it will be forgotten again.

Maybe it would even makes sense to remove the `GITHUB_` prefix then and leave it at `HEAD_SHA`? Otherwise it somehow implies that the variable is somehow defined / provided by GitHub.

The comment could be something along the lines of:

> We can't consistently use the built-in `GITHUB_SHA` variable due to differences in triggers, therefore `HEAD_SHA` should be passed explicitly and either set to `GITHUB_SHA`, or to another appropriate value.

Reply:
> I think, however, and this is what might have confused you, that I should better name the variable HEAD_HASH instead of GITHUB_HEAD_SHA, to indicate that this is a non-standard environment variable that we are setting explicitly.

> Maybe it would even makes sense to remove the GITHUB_ prefix then and leave it at HEAD_SHA? Otherwise it somehow implies that the variable is somehow defined / provided by GitHub

Seems like we were thinking along the same lines! :) How about, 2040deec766606836dc86ae8db8f7d8d36cac793? I'm not sure where you think the comment would be most helpful.